src/lib/swapper/swappers/ZrxSwapper/endpoints.ts

```diff
@@ -0,0 +1,173 @@
+    input: GetTradeQuoteInput,
+    { sellAssetUsdRate }: { sellAssetUsdRate: string },
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    const tradeQuoteResult = await getZrxTradeQuote(
+      input as GetEvmTradeQuoteInput,
+      sellAssetUsdRate,
+    )
+
+    return tradeQuoteResult.map(tradeQuote => {
+      const { receiveAddress, affiliateBps } = input
+      const id = uuid()
+      tradeQuoteMetadata.set(id, { chainId: tradeQuote.steps[0].sellAsset.chainId as EvmChainId })
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<ETHSignTx> => {
+    const { accountNumber, buyAsset, sellAsset, sellAmountBeforeFeesCryptoBaseUnit } =
+      tradeQuote.steps[stepIndex]
+
+    const { receiveAddress, recommendedSlippage, affiliateBps } = tradeQuote
+
+    const chainId = sellAsset.chainId
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(chainId) as ChainAdapter<EvmChainId>
+
+    if (adapter === undefined) {
+      throw new SwapError('[executeTrade] - getChainAdapterManager returned undefined', {
+        code: SwapErrorType.UNSUPPORTED_CHAIN,
+        details: { chainId },
+      })
+    }
+
+    if (!isEvmChainAdapter(adapter)) {
+      throw new SwapError('[executeTrade] - non-EVM chain adapter detected', {
+        code: SwapErrorType.EXECUTE_TRADE_FAILED,
+        details: {
+          chainAdapterName: adapter.getDisplayName(),
+          chainId: adapter.getChainId(),
+        },
+      })
+    }
+
+    const maybeZrxQuote = await fetchZrxQuote({
+      buyAsset,
+      sellAsset,
+      receiveAddress,
+      slippage: recommendedSlippage, // TODO: use the slippage from user input
+      affiliateBps,
+      sellAmountBeforeFeesCryptoBaseUnit,
+    })
+
+    if (maybeZrxQuote.isErr()) throw maybeZrxQuote.unwrapErr()
+    const { data: zrxQuote } = maybeZrxQuote.unwrap()
+
+    const { value, to, gasPrice, gas, data } = zrxQuote
+
+    const buildSendApiTxInput = {
+      value: value.toString(),
+      to,
+      from: from!,
+      chainSpecific: {
+        gasPrice: gasPrice.toString(),
+        gasLimit: gas.toString(),
+        maxFeePerGas: undefined,
+        maxPriorityFeePerGas: undefined,
+        data: data.toString(),
+      },
+      accountNumber,
+    }
+    return adapter.buildSendApiTransaction(buildSendApiTxInput)
+  },
+
+  checkTradeStatus: async ({
+    tradeId,
+    txId,
+  }): Promise<{ status: TxStatus; buyTxId: string | undefined; message: string | undefined }> => {
+    try {
+      const maybeTradeQuoteMetadata = tradeQuoteMetadata.get(tradeId)
+      if (!maybeTradeQuoteMetadata) {
+        return createDefaultStatusResponse(txId)
+      }
+      const baseUrl = getUnchainedBaseUrl(maybeTradeQuoteMetadata.chainId)
+      const {
+        data: { status: rawStatus },
+      } = await axios.get<Tx>(`${baseUrl}/api/v1/tx/${txId}`)
```

Comment:
Fine for now, but what about importing `http as v1HttpApi` from the relevant plugin e.g `plugins/ethereum` and calling `v1HttpApi.getTransaction()` so we don't reinvent the wheel?

The only thing is we'll have to be programmatic and make sure we import the right chain' unchained-client API, which we could do with a mapping of ChainId to `unchained-client` api

Reply:
good idea, also kev recommended we do it via chain adapters to adopt and sdk-like pattern